pkg/common/client/config.go

```diff
@@ -221,6 +225,11 @@ func (b *configBuilder) Build() (*sarama.Config, error) {
 		config.ClientID = b.clientId
 	}
 
+	if ctx != nil {
+		logger := logging.FromContext(ctx)
+		logger.Debugf("Built Sarama config: %+v", config)
```

Comment:
Usually with these zap loggers we would use something like this, I think.  I'm not sure if it makes any practical difference though.
```suggestion
		logger.Debug("Built Sarama config", zap.Any("config", config))
```

Reply:
I never learnt the Zap logger properly, maybe there's a way to do it better but suggested change doesn't work.

That results in this:
```
Built Sarama config{config 23 0  0xc0000bb180}
```

Reply:
Apologies; I didn't test this specific suggestion.  I haven't noticed that much of a difference when I code one way vs. the other and was only noting it for consistency across the eventing codebase.  If it doesn't work here I wouldn't push for it.

In my quick test with a different value, the mechanisms look very similar; I'm not sure why it only printed a pointer value with the sarama config.

Using configMap *corev1.ConfigMap :

Infof("zap test: %v", configMap)  ==>   "zap test: &ConfigMap{ObjectMeta:{config-kafka  knative-eventing /api/v1/namespaces/knative-eventing/..."
Info("zap test", zap.Any("test", configMap))  ==>  "zap test: {configMap 25 0  &ConfigMap{ObjectMeta:{config-kafka  knative-eventing /api/v1/namespaces/knative-eventing/..."

Reply:
Just for curiosity's sake, I noticed this:
```
any := zap.Any("test", saramaConfig)
fmt.Print(any)   // Outputs "{saramaConfig 23 0  0xc00049ea80}"
fmt.Print(any.Interface)   // Outputs "&{{{5 100ms} 3s} {5 30s 30s 30s {false <nil>} ...."
```
Apparently the reflection calls (in fmt/print.go) don't like the zap.Field with a zapcore.ReflectType in it (which zap.Any returns) -- for whatever reason it doesn't delve into unpacking the entire zap.Field interface to get to the zap.Field.Interface where the saramaConfig is actually stashed.

When using zap.Any with a *corev1.ConfigMap, the zap.Field.Type is "StringerType" and not "ReflectType" which causes it to go down a different path.

Reply:
Any changes needed in the code? Sorry wasn't clear in the comment :)

Reply:
No changes; I was just expressing my confusion at the way the zap.Any() call works (or doesn't work, in this case).  Thanks!